weblate/trans/functions.py

```diff
@@ -0,0 +1,50 @@
+# Copyright © Michal Čihař <michal@weblate.org>
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from django.db.models import DateTimeField, Expression, Func
+
+VALID_UNIT_VALUES = {
+    "MICROSECOND",
+    "SECOND",
+    "MINUTE",
+    "HOUR",
+    "DAY",
+    "WEEK",
+    "MONTH",
+    "QUARTER",
+    "YEAR",
+}
+
+
+class MySQLTimestampAdd(Func):
+    function = "TIMESTAMPADD"
+    output_field = DateTimeField()
+
+    def __init__(self, unit: str, interval: Expression, timestamp: Expression):
+        # unit is a string (not a Value/Expression) as mysql/mariadb throws an
+        # error if the unit argument to TIMESTAMPADD is quoted.
+        if unit not in VALID_UNIT_VALUES:
+            msg = f"Invalid unit: {unit}"
+            raise ValueError(msg)
+        self.unit = unit
+        super().__init__(interval, timestamp)
+
+    def as_sql(
+        self,
+        compiler,
+        connection,
+        **kwargs,
+    ):
```

Comment:
Remove extra kwargs:

```suggestion
    def as_sql(self, compiler, connection):
```